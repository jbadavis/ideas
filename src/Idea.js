import React from 'react';

const Idea = ({
  description,
  editing,
  handleUpdateIdea,
  handleDeleteIdea,
  handleEditIdea,
  time,
  title
}) => (
  <div>
    {editing ? (
      <form>Edit</form>
    ) : (
      <div>
        <h2>{title}</h2>
        <h4>{time}</h4>
        <p>{description}</p>
        <button className="edit" onClick={handleEditIdea}>Edit</button>
        <button className="delete" onClick={handleDeleteIdea}>Delete</button>
      </div>
    )}
  </div>
);

export default Idea;
