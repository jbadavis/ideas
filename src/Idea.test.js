import React from 'react';
import {format} from 'date-fns';
import Idea from './Idea';
import {shallow} from 'enzyme';

jest.mock('date-fns');

describe('Idea', () => {
  const handleUpdateIdea = jest.fn();
  const handleDeleteIdea = jest.fn();
  const handleEditIdea = jest.fn();
  const timeMock = "2000-09-24T20:40:51.117Z";

  describe('if editing is truthy', () => {
    it('should match snapshot', () => {
      expect(
        shallow(
          <Idea
            title="title"
            description="description"
            time={timeMock}
            editing={true}
            handleUpdateIdea={handleUpdateIdea}
            handleDeleteIdea={handleDeleteIdea}
            handleEditIdea={handleEditIdea}
          />,
        ),
      ).toMatchSnapshot();
    });
  });

  describe('if editing is falsy', () => {
    const component = shallow(
      <Idea
        title="title"
        description="description"
        time={timeMock}
        editing={false}
        handleUpdateIdea={handleUpdateIdea}
        handleDeleteIdea={handleDeleteIdea}
        handleEditIdea={handleEditIdea}
      />,
    );

    it('should update state', () => {
      expect(component).toMatchSnapshot();
    });

    it('should call date-fns format correctly', () => {
      expect(format).toHaveBeenCalledWith(new Date(timeMock), 'hh:mm aaa d/M/yy');
    });

    describe('if the edit CTA is clicked', () => {
      it('should call handleEditIdea', () => {
        component.find('.edit').simulate('click');

        expect(handleEditIdea).toHaveBeenCalled();
      });
    });

    describe('if the delete CTA is clicked', () => {
      it('should call handleDeleteIdea', () => {
        component.find('.delete').simulate('click');

        expect(handleDeleteIdea).toHaveBeenCalled();
      });
    });
  });
});
