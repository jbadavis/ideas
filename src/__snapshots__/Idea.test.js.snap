// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Idea if editing is falsy should update state 1`] = `
<div>
  <div>
    <h2>
      title
    </h2>
    <h4>
      time
    </h4>
    <p>
      description
    </p>
    <button
      className="edit"
      onClick={[Function]}
    >
      Edit
    </button>
    <button
      className="delete"
      onClick={[Function]}
    >
      Delete
    </button>
  </div>
</div>
`;

exports[`Idea if editing is truthy should match snapshot 1`] = `
<div>
  <AddIdea
    description="description"
    handleUpdateIdea={[Function]}
    title="title"
  />
</div>
`;
